tiingo/api.py

```diff
@@ -155,6 +166,67 @@ def get_ticker_price(self, ticker,
         else:
             return response.content.decode("utf-8")
 
+    def _build_url(self, stock, startDate, endDate, frequency):
+        url = "https://api.tiingo.com/tiingo/"
+        url += "daily/{}/prices?".format(stock)
+        if startDate is not None and endDate is None:
+            url += "&startDate={}".format(startDate)
+        if startDate is not None and endDate is not None:
+            url += "&startDate={}&endDate={}".format(startDate, endDate)
+        url += "&format=json&resampleFreq={}&token={}".format(frequency, self._api_key)
+        return url
+
+    def get_dataframe(self, tickers,
+                      startDate=None, endDate=None, metric_name='adjClose', frequency='daily'):
+
+        """ Return a pandas.DataFrame of historical prices for one or more ticker symbols.
+
+            By default, return latest EOD Composite Price for a list of stock tickers.
+            On average, each feed contains 3 data sources.
+
+            Supported tickers + Available Day Ranges are here:
+            https://apimedia.tiingo.com/docs/tiingo/daily/supported_tickers.zip
+
+            Args:
+                tickers (string/list): One or more unique identifiers for a stock ticker.
+                startDate (string): Start of ticker range in YYYY-MM-DD format.
+                endDate (string): End of ticker range in YYYY-MM-DD format.
+                metric_name (string): Optional parameter specifying metric to be returned for each
+                    ticker.  In the event of a single ticker, this is optional and if not specified
+                    all of the available data will be returned.  In the event of a list of tickers,
+                    this parameter is required.
+                frequency (string): Resample frequency (defaults to daily).
+        """
+
+        valid_columns = ['open', 'high', 'low', 'close', 'volume', 'adjOpen', 'adjHigh', 'adjLow',
+                         'adjClose', 'adjVolume', 'divCash', 'splitFactor']
+        if pandas_is_installed:
+            prices = pd.DataFrame()
+            if type(tickers) is str:
+                stock = tickers
+                url = self._build_url(stock, startDate, endDate, frequency)
+                prices = pd.read_json(url)
+                prices.index = prices['date']
```

Comment:
Nifty pandas trick that's relevant here

`df = pd.read_json(url, index=['date])` will set the date column as the index column, eliminating the need to manually delete/set it.

Reply:
Didn't realize that was a limitation of the read_json method- good to know, I think your approach is a good one.